onnx/defs/tensor/defs.cc

```diff
@@ -371,6 +373,80 @@ ONNX_OPERATOR_SET_SCHEMA(
           ctx.getOutputType(0)->mutable_tensor_type()->mutable_shape();
         }));
 
+static const char* Dim_ver15_doc = R"DOC( 
+Returns a specific dimension of the given input tensor's shape.  
+Takes an input tensor and the index (axis) of the dimension to extract (as an attribute) and
+returns the dimension value at the requested index (axis) in the input shape.
+The op returns the output as a 1-dimensional tensor of size 1,
+for the common usage scenario of creating other shapes (as 1-dimensional tensors).
+The axis specified must be a value in the range [-r, r-1] where r = rank(input).
+Otherwise, the output value is undefined.
+ 
+For example: 
+Input tensor with shape: [2, 3, 4] 
+Index: 1
+Output: [3] 
+ 
+Input tensor with shape: [2, 3, 4] 
+Index: 2
+Output: [4] 
+ 
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    Dim,
+    15,
+    OpSchema()
+        .SetDoc(Dim_ver15_doc)
+        .Input(0, "input", "An input tensor.", "T", OpSchema::Single, true, 1, OpSchema::NonDifferentiable)
+        .Output(
+            0,
+            "output",
+            "Dimension of the input tensor at the specified axis.",
```

Comment:
nit: elaborate this sentence similar to:  Dimension of the given input tensor's shape at the specified axis.

Reply:
Sure. Perhaps something like "The slice of the input tensor's shape at the specified axis"? This started me thinking about the name of the op. The word "dimension" is itself overloaded, since it is also used, like "axis", to indicate a specific axis. But I can't think of a better name for the operator itself. MLIR does use the same opname "Dim", for what it is worth. (We could overload "Shape", making axis an optional parameter ... we can't define it as a function then, but this could be an option.)